src/proto.rs

```diff
@@ -340,6 +341,95 @@ pub struct AllowCredentials {
     pub transports: Option<Vec<String>>,
 }
 
+/// Valid credential protection policies
+#[derive(Debug, Serialize, Clone, Deserialize)]
+#[serde(rename_all = "camelCase")]
+#[repr(u8)]
+pub enum CredentialProtectionPolicy {
+    /// This reflects "FIDO_2_0" semantics. In this configuration, performing
+    /// some form of user verification is optional with or without credentialID
+    /// list. This is the default state of the credential if the extension is
+    /// not specified.
+    UserVerificationOptional = 0x1,
+    /// In this configuration, credential is discovered only when its
+    /// credentialID is provided by the platform or when some form of user
+    /// verification is performed.
+    UserVerificationOptionalWithCredentialIDList = 0x2,
+    /// This reflects that discovery and usage of the credential MUST be
+    /// preceded by some form of user verification.
+    UserVerificationRequired = 0x3,
+}
+
+impl TryFrom<u8> for CredentialProtectionPolicy {
+    type Error = &'static str;
+
+    fn try_from(v: u8) -> Result<Self, Self::Error> {
+        use CredentialProtectionPolicy::*;
+        match v {
+            0x1 => Ok(UserVerificationOptional),
```

Comment:
I need to add a test here because this conversion isn't tightly bound to the actual enum values.